Sources/Datadog/Tracer.swift

```diff
@@ -175,11 +175,15 @@ public class Tracer: OTTracer {
 
     public func extract(reader: OTFormatReader) -> OTSpanContext? {
         // TODO: RUMM-385 - make `HTTPHeadersReader` available in public API
-        guard let reader = reader as? HTTPHeadersReader else {
+        if let reader = reader as? HTTPHeadersReader {
+            reader.use(baggageItemQueue: queue)
+            return reader.extract()
+        } else if let reader = reader as? OpenTelemetryHTTPHeadersReader {
+            reader.use(baggageItemQueue: queue)
+            return reader.extract()
+        } else {
```

Comment:
At a glance wouldn't having a shared protocol for both `HTTPHeadersReader` and `OpenTelemetryHTTPHeadersReader` let us have a single `if` block instead of duplicating code?

Reply:
definitely! I already started introducing this in the follow up PR that will enable automatic selection of encoding. I wanted to avoid touching Open Tracing interfaces in this PR.